modules/data.remote/R/LandTrendr.AGB.R

```diff
@@ -0,0 +1,358 @@
+                            enddate = character(total_rows), 
+                            dbfile.name = "LandTrendr", 
+                            stringsAsFactors = FALSE)
+
+  for (i in seq_len(total_rows)) { 
+    results$file[i] <- downloaded_files[i]
+    results$host[i] <- PEcAn.remote::fqdn()
+    results$startdate[i] <- paste0(downloaded_years[i], "-01-01 00:00:00")
+    results$enddate[i] <- paste0(downloaded_years[i], "-12-31 23:59:59")
+    results$formatname[i] <- out_formats[i]
+  }
+  
+  return(results)
+}
+#
+
+#
+##' @title extract.LandTrendr.AGB
+##' @name  extract.LandTrendr.AGB
+##' 
+##' @param coords vector of BETYdb site IDs or a data frame containing elements 'lon' and 'lat'
+##' @param buffer Optional. operate over desired buffer area [not yet implemented]
+##' @param fun Optional function to apply to buffer area.  Default - mean
+##' @param data_dir  directory where input data is located. Can be NUL if con is specified
+##' @param product_dates Process and extract data only from selected years. Default behavior
+##' (product_dates = NULL) is to extract data from all availible years in BETYdb or data_dir
+##' @param con connection to PEcAn database. Can be NULL if data_dir is specified
+##' @param output_file Path to save LandTrendr_AGB_output.RData file containing the 
+##' output extraction list (see return)
+##' @param plot_results Optional. Create a simple diagnostic plot showing results of 
+##' data extraction by site/coordinate pair
+##' 
+##' @return list of two containing the median AGB values per pixel and the corresponding 
+##' standard deviation values (uncertainties)
+##' 
+##' ##' @examples
+##' \dontrun{
+##' 
+##' # Example 1 - using BETYdb site IDs to extract data
+##' # Database connection (optional)
+##' bety <- list(user='bety', password='bety', host='localhost',
+##' dbname='bety', driver='PostgreSQL',write=TRUE)
+##' con <- PEcAn.DB::db.open(bety)
+##' bety$con <- con
+##' 
+##' data_dir <- "~/scratch/agb_data/"
+##' site_ID <- c(2000000023,1000025731,676,1000005149) # BETYdb site IDs
+##' results <- PEcAn.data.remote::extract.LandTrendr.AGB(coords=site_ID, 
+##'            data_dir = data_dir, con = con, 
+##'            output_file = file.path(data_dir))
+##'            
+##' 
+##' # Example 2 - provide arbitrary lon/lat values
+##' coords <- data.frame(lon=c(-90.8,-91.8), lat=c(42.5,43.5))
+##' product_dates <- c(1990,1991)
+##' results <- PEcAn.data.remote::extract.LandTrendr.AGB(coords=coords, 
+##'            data_dir = data_dir, product_dates = product_dates,
+##'            output_file = file.path(data_dir)
+##' }
+##' 
+##' @export
+##' @author Shawn Serbin, Alexey Shiklomanov
+##' 
+extract.LandTrendr.AGB <- function(coords, buffer = NULL, fun = "mean", data_dir = NULL, 
+                                   con = NULL, product_dates = NULL,
+                                   output_file = NULL, 
+                                   plot_results = FALSE, ...) {
```

Comment:
Going to keep this high level:
1. The crux of the issue with GeoTIFF seems to be the geographic projection. Does netCDF support projections? If so, I think this function is unnecessary -- check if Brown Dog DAP supports GeoTIFF -> netCDF and if not work with @robkooper to get it in.
2. If we do need to work with GeoTIFF, then this function should conform to load_data's expectations: `load_[mime_type](data.path, format, site, vars = NULL)`. As far as I can tell there's nothing below that requires that this function be implemented on a format-specific basis rather than a mime_type basis.
3. Internal inclusion of median and stdev cases is redundant. Generalize and just call the function twice
4. Assume site info is passed in as a list with elements (id, lat, lon, time_zone). Don't pass `con` in as this prevents the function from running remote
5. plot_results is just diagnostic, punt it to it's own function and call it later
6. if projection is file specific, we need to make sure that info is passed in with the `format` record.

Reply:
@mdietze OK so far to answer some of these:

1. Yes, I have played with this and we can use gdal_translate to convert o netCDF with projection info and then read that info in R and extract points from the netCDF instead of GeoTIFF based on lat/long etc.

Right now BD does not support large files or netCDF . I was thinking of invoking gdal to convert geoTIFF to netcdf and then storing these in the DB

2. the load_data convention is there, but I need to build out the database side and then register the files - just didnt do this yet until I was sure I was ready

3. OK

4. OK.  I will switch from DB to a list with the needed elements.  Do we have an example of creating this list of info from the DB as a generic function?

5. OK I will pull out to a separate function to call after

6.  OK so you mean dont read project from the file but put that CRS info into the format record? Do we have an example of this somewhere?

Reply:
@mdietze I have made updates this these functions in the hopes of getting them merged for SDA testing.  I will then work on modifying to work with "formats" and "load data" since it seems for now we will be sticking with tiff for now.  If we can merge this in the interim, when happy with it, we can use it with our existing multi-site SDA testing easier than source this code from the script.

@bailsofhay @para2x the changes I have made to this will "break" your current SDA scripts insofar as you now need to read in two different files for median and stdv and no longer pass a con but instead a new site_info list.  

Finally, I think a use can still pass generic lat/longs to the function they just do it in the site_info list and either provide custom site names or NAs.

Thoughts on these changes?  Again, I know we ultimately need to integrate with BETY